services/galley/src/Galley/Data/Types.hs

```diff
@@ -113,5 +113,5 @@ generate cnv s t = do
 
 mkKey :: MonadIO m => ConvId -> m Key
 mkKey cnv = do
-    Just sha256 <- liftIO $ getDigestByName "SHA256"
+    sha256 <- liftIO $ fromJust <$> getDigestByName "SHA256"
```

Comment:
This change makes error messages a lot worse, at least historically.  What's the motivation to avoid the incomplete pattern instead?  Does that have something to do with ghc upgrade and `MonadFail`?

Reply:
Yes, adding `MonadFail` here was a more _hairy_ change. But since this should never error, we don't have to worry about bad error messsages.